### PR TITLE
Muon - Core Code; New Load Utilities

### PR DESCRIPTION
**Pull request by TWJubb**
Refs #23642 

~~Requires #23644 to be merged.~~

This PR extends the `load_utils.py` utility file with several useful functions which are used frequently in the new interface.

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/load_utils.py [hunk 1]
```diff
@@ -84,10 +101,387 @@ def getWorkspaceNames(self):
...
+
+
+def get_default_instrument():
+    default_instrument = ConfigServiceImpl.Instance().getInstrument().name()
+    if default_instrument not in file_utils.allowed_instruments:
+        default_instrument = "None"
+    return default_instrument
+
+
+def run_LoadInstrument(parameter_dict):
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/load_utils.py]: having an alg in this file does not feel right to me

File: scripts/Muon/GUI/Common/load_utils.py [hunk 2]
```diff
@@ -84,10 +101,387 @@ def getWorkspaceNames(self):
...
+                   "MainFieldDirection"]
+# List of default values for the DEFAULT_OUTPUTS list
+DEFAULT_OUTPUT_VALUES = [__default_workspace(),
+                         None,  # api.WorkspaceFactoryImpl.Instance().createTable("TableWorkspace"),
+                         api.WorkspaceFactoryImpl.Instance().createTable("TableWorkspace"),
+                         0.0,
+                         0.0, "Unknown direction"]
+
+
+class floatPropertyWithValue(object):
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/load_utils.py]: do you know where this is used? I dont understand its purpose from the description

File: scripts/Muon/GUI/Common/load_utils.py [hunk 3]
```diff
@@ -84,10 +101,387 @@ def getWorkspaceNames(self):
...
+    filename = alg.getProperty("Filename").value
+
+    return load_result, run, filename
+
+
+def empty_loaded_data():
+    return dict(zip(DEFAULT_OUTPUTS + ["DataDeadTimeTable"], DEFAULT_OUTPUT_VALUES + [None]))
+
+
+def create_load_algorithm(filename, property_dictionary):
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/load_utils.py]: I think this should be somewhere else

File: scripts/Muon/GUI/Common/load_utils.py [hunk 4]
```diff
@@ -84,10 +101,387 @@ def getWorkspaceNames(self):
...
+    Return a list of names of TableWorkspace objects which are in the ADS.
+    """
+    names = api.AnalysisDataService.Instance().getObjectNames()
+    table_names = [name for name in names if isinstance(mtd[name], ITableWorkspace)]
+    return table_names
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Saving/Loading grouping XML
+# ----------------------------------------------------------------------------------------------------------------------
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/load_utils.py]: lets split these off into a new file

File: scripts/Muon/GUI/Common/load_utils.py [hunk 5]
```diff
@@ -84,10 +101,387 @@ def getWorkspaceNames(self):
...
+        if child.tag == "pair":
+            names += [child.attrib['name']]
+            groups += [[child.find('forward-group').attrib['val'], child.find('backward-group').attrib['val']]]
+            alphas += [child.find('alpha').attrib['val']]
+    return names, groups, alphas
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Mantid algorithms
+# ----------------------------------------------------------------------------------------------------------------------
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/load_utils.py]: these should be split off too

**Comment by AnthonyLim23** [comment]
I think these files should be in some sort of utils folder

File: scripts/Muon/GUI/Common/load_utils.py (hunk 2 above)
Comment (Mantid-Matthew) [reply scripts/Muon/GUI/Common/load_utils.py]: This is I think used when initializing with default values in MuonLoader. I think it is used to make a dummy workspace look like a workspace.

Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/load_utils.py]: could we give it a better description? I think the default values should come from the context.

Comment (Mantid-Matthew) [reply scripts/Muon/GUI/Common/load_utils.py]: Having double checked I can't find anywhere where it is currently used. I thought I had found a usage but it doesn't seem to be there. I will remove for now.

**Comment by AnthonyLim23** [comment]
Im happy once the builds pass

**Comment by AnthonyLim23** [comment]
I have found a bug:

- open muon analysis and load some data
- open frequency domain analysis
- you will get the warning ```
Error in execution of algorithm LoadInstrument
Failed to find an instrument with this name in any facility: 'None' - search object None
```

**Review by AnthonyLim23 — APPROVED** [review]
all works now. @mantidproject/gatekeepers  the build failures are unrelated

**Comment by martyngigg** [comment]
Confirmed that the tests are the unreliable `KafkaHistoListener` test.

**Pull request merged**